clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -268,11 +268,7 @@ public void updateEndOffsets(Map<TopicPartition, Long> newOffsets) {
     @Override
     public List<PartitionInfo> partitionsFor(String topic) {
         ensureNotClosed();
-        List<PartitionInfo> parts = this.partitions.get(topic);
-        if (parts == null)
-            return Collections.emptyList();
-        else
-            return parts;
+        return this.partitions.get(topic);
```

Comment:
Even if returning null for unknown topic is the original behavior of KafkaConsumer?
https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java#L1238
https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java#L233

At the moment I confirmed that only the `ProcessorStateManagerTest#testNoTopic` turns into fail by this change, which is fixed by the commit f8d9620.
In order to prevent another test unexpectedly pass due to this difference between KafkaConsumer and MockConsumer, I still think it make sense to fix MockConsumer, but let me know if you still have concerns to do this.